### DISCUSSION TITLE
Callsign validity checker command

### DISCUSSION THREAD

**Discussion by classabbyamp** [miaowware/qrm2#402] [Ideas]
> I'm half-thinking that we should make a callsign validator when the command is provided with a call after the country

```
>>> ?pfx us
[us table]
>>> ?pfx us N0CALL
Invalid format: 1x4
```

> does that means that pfx is a group or we parse that as an optional arg?

Could use my callsign.me project as a jumping off point (requires finishing callsign.me)

**Comment by 0x5c** [comment]
But callsign.me does not do simple validation, unless I misunderstood
And a request to an api for that seems a bit futile to me

**Comment by classabbyamp** [comment]
the intent of the project is to do validation, so that code could be re-used for the bot. no api request.

**Comment by 0x5c** [comment]
ah ok, that makes sense

**Comment by classabbyamp** [comment]
ITU callsign rules: https://www.itu.int/en/ITU-R/terrestrial/fmd/Documents/fxm-art19-sec3.pdf

**Comment by classabbyamp** [comment]
I have a callsign regex here: `([0-9A-Z]+/)?([BFGIKMNRW]|[0-9A-Z]{2,3})[0-9][0-9A-Z]*[A-Z](/[0-9A-Z]+)?` that should work for matching callsigns, including `PFX/CALL/SFX`

**Comment by 0x5c** [comment]
That regex could also capture the groups and allow for the result to have info about the parts, ie country prefix, area number, suffix, if the call is special long format, etc
